src/UIAutomation/FunctionalTests/UiaOperationAbstractionTests.cpp

```diff
@@ -2715,5 +2715,113 @@ namespace UiaOperationAbstractionTests
+                try
+                {
+                    scope.Resolve();
+                }
+                catch(winrt::hresult_error& e)
+                {
+                    Assert::AreEqual(E_FAIL, static_cast<HRESULT>(e.code()));
+                    throw;
+                }
+            });
+
+            // And we can still access the calculated results so far
+            Assert::AreEqual(static_cast<int>(x) > 0, true);
+            LogOutput(L"x = ",static_cast<int>(x));
+        }
+
+        // Test that  a runtime error within a remote operation causes an UnhandledRemoteException in resolve.
+        TEST_METHOD(unhandledRemoteException)
+        {
+            auto guard = InitializeUiaOperationAbstraction(true);
+
+            ModernApp app(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app.Activate();
+            auto calc = WaitForElementFocus(L"Display is 0");
+
+            auto scope = UiaOperationScope::StartNew();
+
+            UiaElement element = calc;
+
+            UiaArray<UiaInt> a;
+            // Accessing an element of an empty array should cause an E_BOUNDS error
+            auto x = a.GetAt(0);
+
+            Assert::ExpectException<UnhandledRemoteException>([&]()
+            {
+                try
+                {
+                    scope.Resolve();
+                }
+                catch(winrt::hresult_error& e)
+                {
+                    Assert::AreEqual(E_BOUNDS, static_cast<HRESULT>(e.code()));
+                    throw;
+                }
+            });
+        }
+
+        // Test that  a execution failure within a remote operation causes an ExecutionFailureException in resolve.
+        TEST_METHOD(executionFailureException)
+        {
+            auto guard = InitializeUiaOperationAbstraction(true);
+
+            ModernApp app1(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app1.Activate();
+            auto calc1 = WaitForElementFocus(L"Display is 0");
+
+            ModernApp app2(L"Microsoft.WindowsCalculator_8wekyb3d8bbwe!App");
+            app2.Activate();
+            auto calc2 = WaitForElementFocus(L"Display is 0");
+            
+            auto scope = UiaOperationScope::StartNew();
+
+            UiaElement element1 = calc1;
+            UiaElement element2 = calc2;
+            UiaBool sameNames = element1.GetName() == element2.GetName();
+            scope.BindResult(sameNames);
+
+            // As this operation deals with elements from 2 different processes
+            // resolving will cause an execution failure.
+            Assert::ExpectException<ExecutionFailureException>([&]()
+            {
+                try
+                {
+                    scope.Resolve();
+                }
+                catch(winrt::hresult_error& e)
+                {
+                    Assert::AreEqual(E_UNEXPECTED, static_cast<HRESULT>(e.code()));
+                    throw;
+                }
+            });
+        }
```

Comment:
We can produce malformed bytecode errors by passing over 1MB of bytecode to the remote process. Could we add a test for that? It should be possible to do that similarly to the instruction limit test, but with an actual C++ loop:

```cpp
for (int i = 0; i < 1048576; ++i)
{
    UiaBool b = true;
}
```

We definitely don't need to iterate this many times (the `NewBool` instruction plus its operands is a total of 9 bytes), but this is the basic idea.